Week3/homework/app.js

```diff
@@ -0,0 +1,150 @@
+const express = require('express');
+const mysql = require('mysql');
+const bodyParser = require('body-parser');
+const app = express();
+
+app.use(bodyParser.json());
+app.use(express.urlencoded({ extended: false }));
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'todo',
+});
+
+// Create User
+const createUser = (req, res) => {
+  const user_name = req.body.name;
+  if (!user_name) {
+    return res.status(400).send('Please enter a username for user!');
+  }
+  const createUser = `INSERT INTO Users (username) VALUES (?);`;
+  connection.query(createUser, [user_name], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+  });
+  return res.status(201).send('New user successfully created!');
+};
+
+//Get users
+const getUsers = (req, res) => {
+  const getUsersQuery = `SELECT * from users`;
```

Comment:
If this was a production database, we should try to avoid doing SELECTs like this without limits because if we had billions of users, we don't generally want to do this. It might break the page, or it can be very slow!